content/docs/rfcs/72/README.md

```diff
@@ -0,0 +1,273 @@
+
+## Example Waku RLN Keystore:
+
+This is an example of a keystore used by a [17/WAKU2-RLN-Relay](https://rfc.vac.dev/spec/17/).
+
+```js
+application: 1 ,
+appIdentifier: 2,
+version: 3,
+credentials: {
+    membershipHash: {
+      crypto: {
+        cipher: 4,
+        cipherparams: 5,
+        ciphertext: 6,
+        kdf: 7,
+        kdfparams: {
+          dklen: 8,
+          c: 9,
+          prf: 10,
+          salt: 11,
+        },
+        mac: 12,
+      }
+    }
+}
+
+```
+
+# Specification
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “NOT RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in [RFC 2119](https://datatracker.ietf.org/doc/html/rfc2119) and [RFC 8174](https://datatracker.ietf.org/doc/html/rfc8174).
+#
+
+The keystore MUST be generated with a cryptographic construction for password verification and decryption.
+
+Keystore modules MUST include metadata, key derivation function, checksum, cipher, and a membership hash.
+
+
+## Metadata:
+- Information about the keystore SHOULD be stored in the metadata. 
+- The declaration of `application`, `version`, and `appIdentifier` COULD occur in the metadata.
+
+`application` : current application, MUST be a string
+
+`version` : application version, MUST be a string
+
+`appIdentifier`: application identifier, MUST be a string
+
+## Credentials:
+The Waku RLN credentials MUST consist of a `membershipHash` and `WakuCredential`
+- Each contruct MUST include the keypair:
+> key: [membershipHash]: pair: [WakuCredential]
+
+### membershipHash 
+- MUST be a 256 byte hash.
+- it Must be a string
+- SHOULD be generated with `treeIndex`, `membershipContract`, and `identityCredential`.
+- MUST not already exist in the keystore.
+
+`treeIndex`  
+- it MUST be a RLN membership tree index in a merkle tree data structure filled with `identity_commitment` from user registrations.
+As described in [32/RLN-V1](https://rfc.vac.dev/spec/32/)
+- MUST be integer
+
+`membershipContract` 
+- it MUST be a hash of a `contractId` and `contractAddress`
+- `contractId` MUST be an integer.
+- `contractAddess` MUST be a string.
+
+`identityCredential` 
+- it MUST be a hash of `identity_commitment` stored in a Merkle tree.
+- MUST be a string.
+
+it MUST consists of:
+- `identity_secret`= `identity_nullifier` + `identity_trapdoor` 
+  - `identity_nullifier` : Random 32 byte value
+  - `identity_trapdoor` : Random 32 byte value
+
+`identity_secret_hash` 
+- it MUST be created with `identity_secret` as a parameter for the hash function.
+- Used to decrypt the `identity_commitment` of the user, and as
+a private input for zero-knowledge proof generation.
+- The secret hash SHOULD be kept private by the user.
+
+`identity_commitment` 
+- it MUST be created with `identity_secret_hash` for hash creation. 
```

Comment:
can you describe how it is created?
please omit "hash creation"

Reply:
also made this a bit more descriptive, @rymnc